velox/functions/lib/Re2Functions.cpp

```diff
@@ -300,6 +342,43 @@ class Re2SearchAndExtract final : public VectorFunction {
   }
 };
 
+class LikeConstantPattern final : public VectorFunction {
+ public:
+  explicit LikeConstantPattern(StringView pattern, char escapeChar)
+      : re_(toStringPiece(likePatternToRe2(pattern, escapeChar)), RE2::Quiet) {}
```

Comment:
You need to store pattern string itself in a member variable. Otherwise, when pattern goes out of scope the memory pointed to by toStringPiece gets freed.

Reply:
This is an edge case, but -- you will throw an exception here at instantiation time if the pattern is invalid even if the expression is never executed.

If the pattern was not constant (i.e., we did not optimize the constant case) this would not occur.

@mbasmanova The re2 library stores a copy of the pattern; you don't to keep your own copy.

Reply:
This usage seemed consistent with the rest of the functions in this file.

@funrollloops clarified that Re2 stores the pattern string.